ksql-engine/src/main/java/io/confluent/ksql/util/SerDeUtil.java

```diff
@@ -54,6 +55,20 @@ private static Schema getKSQLSchemaForAvroSchema(org.apache.avro.Schema avroSche
       case MAP:
         return SchemaBuilder.map(Schema.STRING_SCHEMA,
                                  getKSQLSchemaForAvroSchema(avroSchema.getValueType()));
+      case UNION:
+        List<org.apache.avro.Schema> schemaList = avroSchema.getTypes();
+        if (schemaList.size() == 1) {
+          return getKSQLSchemaForAvroSchema(schemaList.get(0));
+        } else if (schemaList.size() == 2) {
+          if (schemaList.get(0).getType() == org.apache.avro.Schema.Type.NULL) {
+            return getKSQLSchemaForAvroSchema(schemaList.get(1));
+          } else if (schemaList.get(1).getType() == org.apache.avro.Schema.Type.NULL) {
+            return getKSQLSchemaForAvroSchema(schemaList.get(0));
+          }
+        }
+        throw new KsqlException(String.format("Cannot find correct type for avro type: %s",
```

Comment:
In this case it would help to have a more informative error message about the restrictions we place on unions:
(at most 2 types, one must be null)